.github/workflows/build.yml

```diff
@@ -80,7 +80,10 @@ jobs:
         run: nix run -f ./nix gomod2nix
       - name: check working directory is clean
         id: changes
-        uses: UnicornGlobal/has-changes-action@v1.0.11
```

Comment:
it seems the command used is pretty simple 
https://github.com/UnicornGlobal/has-changes-action/blob/v1.0.11/entrypoint.sh

Is it what you are trying to do?

Reply:
yeah, similar, but `git diff` can show the actual differences

Reply:
the github action doesn't work for some weird error, you can see in the old run, it has been malfunctioning for a while I think.

Reply:
but `git diff` don't check for untracked files, but I guess it's fine for our use cases.